### PR TITLE
Capsule Compontent and Box component 

### PR DESCRIPTION
**Pull request by NelsonBrandela**
(In Progress) A box and capsule component used in our game to see if there is a collision

### CODE REVIEW

**Comment by kevin-kruusi** [comment]
👍  Try to explain a bit more in your pull request, to give context to anyone who may be reading. Templates can be added to the github project, but you can format your own with github markup language in your description box. 

For [reference](https://embeddedartistry.com/blog/2017/8/4/a-github-pull-request-template-for-your-projects#my-template).

**Comment by k-kruusi** [comment]
vs/ProjectAlteracValley/v15/ipch/d2b004c1d461d370.ipch  I don't think you need to push changes to this file to the master branch.

File: engine/core/entity_component/component/Box_Component.h [hunk 1]
```diff
@@ -3,37 +3,54 @@
 
 #include "../i_Component.h"
 #include "../i_game_object.h"
+#include "Capsule_Component.h"
```
Comment (k-kruusi) [inline engine/core/entity_component/component/Box_Component.h]: Why would Box_Component have an include on Capsule_component? Is this line needed?

**Comment by kevin-kruusi** [comment]
The .ipch files are generated files only there when you have visual studio open.

File: engine/core/entity_component/component/Box_Component.h (hunk 1 above)
Comment (NelsonBrandela) [reply engine/core/entity_component/component/Box_Component.h]: Sorry sir was super busy this week, I include this file in order for me to be able to pass my capsule struct i created as a parameter in a method for box_component and didn't know if there was another way i could do this without having it as an include

**Comment by NelsonBrandela** [comment]
Kk i'll make sure to change that up thanks sir!

**Pull request closed without merging**